python/lsst/pipe/tasks/detectAndMeasure.py

```diff
@@ -0,0 +1,255 @@
+
+    @pipeBase.timeMethod
+    def run(self, exposure, exposureIdInfo, background=None, allowApCorr=True):
+        """!Detect, deblend and perform single-frame measurement on sources and refine the background model
+
+        @param[in,out] exposure  exposure to process. Background must already be subtracted
+            to reasonable accuracy, else detection will fail.
+            The background model is refined and resubtracted.
+            apCorrMap is set if measuring aperture correction.
+        @param[in]     exposureIdInfo  ID info for exposure (an lsst.daf.butlerUtils.ExposureIdInfo)
+        @param[in,out] background  background model to be modified (an lsst.afw.math.BackgroundList),
+            or None to create a new background model
+        @param[in] allowApCorr  allow measuring and applying aperture correction?
+
+        @return pipe_base Struct containing these fields:
+        - exposure: input exposure (as modified in the course of runing)
+        - sourceCat: source catalog (an lsst.afw.table.SourceTable)
+        - background: background model (input as modified, or a new model if input is None);
+            an lsst.afw.math.BackgroundList
+        """
+        if background is None:
+            background = BackgroundList()
+
+        sourceIdFactory = IdFactory.makeSource(exposureIdInfo.expId, exposureIdInfo.unusedBits)
+
+        table = SourceTable.make(self.schema, sourceIdFactory)
+        table.setMetadata(self.algMetadata)
+
+        detRes = self.detection.run(table=table, exposure=exposure, doSmooth=True)
+        sourceCat = detRes.sources
+        if detRes.fpSets.background:
+            background.append(detRes.fpSets.background)
+
+        if self.config.doDeblend:
+            self.deblend.run(
+                exposure = exposure,
+                sources = sourceCat,
+                psf = exposure.getPsf(),
+            )
+
+        self.measure(
+            exposure = exposure,
+            exposureIdInfo = exposureIdInfo,
+            sourceCat = sourceCat,
+            allowApCorr = allowApCorr,
+        )
+
+        return pipeBase.Struct(
+            exposure = exposure,
+            sourceCat = sourceCat,
+            background = background,
+        )
+
+    def measure(self, exposure, exposureIdInfo, sourceCat, allowApCorr=True):
+        """Measure sources
+
+        @param[in,out] exposure  exposure to process. Background must already be subtracted
+            to reasonable accuracy, else detection will fail.
+            Set apCorrMap if measuring aperture correction.
+        @param[in]     exposureIdInfo  ID info for exposure (an lsst.daf.butlerUtils.ExposureIdInfo)
+        @param[in,out] background  background model to be modified (an lsst.afw.math.BackgroundList),
+            or None to create a new background model
+        @param[in] allowApCorr  allow measuring and applying aperture correction?
+        """
+        if self.config.doMeasureApCorr and allowApCorr:
```

Comment:
How about removing some of the unnecessary indentation:

    if not self.config.doMeasureApCorr or not allowApCorr:
        self.measurement.run(...)
        return
    # perform measurements before aperture correction
    self.measurement.run(...)
    # and so on